sw/device/silicon_creator/mask_rom/sigverify_keys_unittest.cc

```diff
@@ -26,15 +29,141 @@ using ::testing::NotNull;
+        .key_type = kSigverifyKeyTypeProd,
+    },
+    {
+        .key = {.n = {{0xC1}}, .n0_inv = {0}, .exponent = 0},
+        .key_type = kSigverifyKeyTypeDev,
+    },
+};
+
+/**
+ * Returns the indices of mock keys of the given type.
+ *
+ * @param key_type A key type.
+ * @return Indices of mock keys of the given type.
+ */
+std::vector<size_t> MockKeyIndicesOfType(sigverify_key_type_t key_type) {
+  std::vector<size_t> res;
+  auto begin = std::begin(kMockKeys);
+  auto end = std::end(kMockKeys);
+  auto iter = begin;
+  while (
+      (iter = std::find_if(iter, end, [&](sigverify_mask_rom_key_t const &key) {
+         return key.key_type == key_type;
+       })) != end) {
+    res.push_back(std::distance(begin, iter));
+    ++iter;
+  }
+  return res;
+}
+
+const std::vector<size_t> kMockTestKeyIndices =
+    MockKeyIndicesOfType(kSigverifyKeyTypeTest);
+
+const std::vector<size_t> kMockProdKeyIndices =
+    MockKeyIndicesOfType(kSigverifyKeyTypeProd);
+
+const std::vector<size_t> kMockDevKeyIndices =
+    MockKeyIndicesOfType(kSigverifyKeyTypeDev);
+
+const std::vector<size_t> kMockAllKeyIndices = []() {
+  std::vector<size_t> res(
+      std::distance(std::begin(kMockKeys), std::end(kMockKeys)));
+  std::iota(res.begin(), res.end(), 0);
+  return res;
+}();
+
+/**
+ * Life cycle states used in parameterized tests.
+ */
+
+const std::vector<lifecycle_state_t> kLcStatesTest{
+    kLcStateTestUnlocked0, kLcStateTestUnlocked1, kLcStateTestUnlocked2,
+    kLcStateTestUnlocked3, kLcStateTestUnlocked4, kLcStateTestUnlocked5,
+    kLcStateTestUnlocked6, kLcStateTestUnlocked7,
+};
+
+const std::vector<lifecycle_state_t> kLcStatesNonTestOperational{
+    kLcStateDev,
+    kLcStateProd,
+    kLcStateProdEnd,
+    kLcStateRma,
+};
+
+const std::vector<lifecycle_state_t> kLcStatesNonOperational{
+    kLcStateRaw,         kLcStateTestLocked0,
+    kLcStateTestLocked1, kLcStateTestLocked2,
+    kLcStateTestLocked3, kLcStateTestLocked4,
+    kLcStateTestLocked5, kLcStateTestLocked6,
+    kLcStateScrap,       kLcStatePostTransition,
+    kLcStateEscalate,    kLcStateInvalid,
+};
+
+const std::unordered_set<lifecycle_state_t> kLcStatesAll = []() {
+  std::unordered_set<lifecycle_state_t> res;
+  res.insert(kLcStatesTest.begin(), kLcStatesTest.end());
+  res.insert(kLcStatesNonTestOperational.begin(),
+             kLcStatesNonTestOperational.end());
+  res.insert(kLcStatesNonOperational.begin(), kLcStatesNonOperational.end());
+  return res;
+}();
+
+TEST(LcStateCount, IsCorrect) {
+  EXPECT_EQ(kLcStateNumStates, kLcStatesAll.size());
+}
+
+class SigverifyKeys : public mask_rom_test::MaskRomTest {
  protected:
+  /**
+   * Sets expectations for getting the keys stored in the Mask ROM.
+   *
+   * @param keys An array of keys.
+   */
+  template <size_t N>
+  void ExpectKeysPtrGet(const sigverify_mask_rom_key_t (&keys)[N]) {
+    size_t num_keys = std::distance(std::begin(keys), std::end(keys));
```

Comment:
Isn't `num_keys` just equal to `N`?

Reply:
Good catch! This wasn't a template initially and forgot to remove this. Done.